lib/linguist/language.rb

```diff
@@ -150,31 +150,25 @@ def self.find_by_filename(filename)
 
     # Public: Look up Languages by file extension.
     #
-    # query - Either a filename String, or an Array of extension strings
```

Comment:
Can you squash the a5f66324eb1b85021bcb57618d5a143150774bc6 commit and it's revert?

Reply:
Why? Everything's going to get squash-merged anyway…

Reply:
So as to not waste the time of reviewers who review commits one by one. I'd also really like if @lildude did not squash all commits of this PR. It makes sense to squash for simple PRs that only add an extension or a language, but not for larger PR such as this one where the different commits can be useful when we git blame later or if we need to revert something.

Reply:
Following that logic, it would make sense to *keep* a5f6632 in case we have need of it later…

Reply:
Following that latest logic, Linguist would be full of unused code :-p

I don't see the point of adding code we're not using. It's only likely to add complexity and bugs.

Reply:
Then perhaps we should keep this simple and *not* deviate from our usual procedure of squash-merging pull-requests. If parts of this need to be reverted later (which is already unlikely), the commits made in this PR can be used as reference.

I don't understand why you're making this more complicated than it needs to be...

Reply:
I think we can meet halfway with a new file. See https://github.com/github/linguist/pull/4936#issuecomment-675875267